src/main/java/com/sap/broker/budgie/resources/api/ServiceInstancesResource.java

```diff
@@ -38,69 +41,274 @@ public ServiceInstancesResource(ServiceBroker serviceBroker) {
+        if (asyncOperation == null) {
+            return emptyBodyResponse(Status.BAD_REQUEST);
+        }
+        return Response.ok(asyncOperation.getOperationState()).build();
+    }
+
+    @GET
+    @Path("/{id}/service_bindings/{binding_id}")
+    public Response getBinding(@PathParam("configId") String configId, @PathParam("id") UUID id, @PathParam("binding_id") UUID bindingId) {
+        ServiceInstance serviceInstance = serviceBroker.get(id, false);
+        if (serviceInstance == null) {
+            return emptyBodyResponse(Status.NOT_FOUND);
+        }
+        Binding binding = serviceInstance.getBinding(bindingId);
+        if (binding == null) {
+            return emptyBodyResponse(Status.NOT_FOUND);
+        }
+        return Response.ok(binding).build();
+    }
+
+    @PUT
+    @Path("/{id}/service_bindings/{binding_id}")
+    public Response bind(@PathParam("configId") String configId, @PathParam("id") UUID id, @PathParam("binding_id") UUID bindingId,
+        @QueryParam("accepts_incomplete") boolean acceptIncomplete, Binding binding) {
+        ServiceInstance serviceInstance = serviceBroker.get(id, false);
+        if (serviceInstance == null) {
+            return emptyBodyResponse(Status.BAD_REQUEST);
+        }
+        if (behaviorEngine.isAsync(configId)) {
+            if (!acceptIncomplete) {
+                return emptyBodyResponse(422);
+            }
+            return bindAsync(configId, bindingId, serviceInstance, binding);
+        }
+        return bindSync(configId, bindingId, serviceInstance, binding);
+    }
+
+    @DELETE
+    @Path("/{id}/service_bindings/{binding_id}")
+    public Response unbind(@PathParam("configId") String configId, @PathParam("id") UUID id, @PathParam("binding_id") UUID bindingId,
+        @QueryParam("accepts_incomplete") boolean acceptIncomplete) {
+        ServiceInstance serviceInstance = serviceBroker.get(id, false);
+        if (serviceInstance == null) {
+            return emptyBodyResponse(Status.BAD_REQUEST);
+        }
+        if (behaviorEngine.isAsync(configId)) {
+            if (!acceptIncomplete) {
+                return emptyBodyResponse(422);
+            }
+            return unbindAsync(configId, bindingId, serviceInstance);
+        }
+        return unbindSync(configId, bindingId, serviceInstance);
+    }
+
+    private Response createSync(String configId, UUID id, ServiceInstance serviceInstance) {
+        sleepIfWanted(configId);
+        Optional<Integer> optionalStatusCode = behaviorEngine.shouldOperationFail(configId, FailConfiguration.OperationType.CREATE, serviceInstance);
+        if (optionalStatusCode.isPresent()) {
+            return emptyBodyResponse(optionalStatusCode.get());
+        }
         serviceInstance.setId(id);
         return create(serviceInstance);
     }
 
+    private Response createAsync(String configId, UUID id, ServiceInstance serviceInstance) {
+        AsyncOperation asyncOperation = new AsyncOperation(() -> {
```

Comment:
extract this runnable creation into another method for better reading
Even better, you would be able to reuse  the newly extracted method